controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -784,7 +784,7 @@ func getMachineNode(ctx context.Context, crClient client.Client, machine *cluste
 		node,
 	)
 	if err != nil {
-		if apierrors.IsNotFound(errors.Cause(err)) {
+		if apierrors.IsNotFound(errors.Cause(err)) || apierrors.IsTimeout(errors.Cause(err)) {
```

Comment:
If we short-circuit after attempting to retrieve the remote client, can we remove the need to check for `apierrors.IsTimeout` here? I worry that the semantics here are quite a bit different, since the `apierrors.IsNotFound` here is seeing if the api server is telling us that the node does not exist.

Reply:
okay, I'll remove. Thank you